spec.html

```diff
@@ -15377,11 +15393,52 @@ <h1>Runtime Semantics: Evaluation</h1>
+      <emu-alg>
+        1. Let _lref_ be the result of evaluating |LeftHandSideExpression|.
+        1. Let _lval_ be ? GetValue(_lref_).
+        1. Let _lbool_ be ! ToBoolean(_lval_).
+        1. If _lbool_ is *true*, return _lval_.
+        1. If IsAnonymousFunctionDefinition(|AssignmentExpression|) and IsIdentifierRef of |LeftHandSideExpression| are both *true*, then
+          1. Let _rval_ be NamedEvaluation of |AssignmentExpression| with argument GetReferencedName(_lref_).
+        1. Else,
+          1. Let _rref_ be the result of evaluating |AssignmentExpression|.
+          1. Let _rval_ be ? GetValue(_rref_).
+        1. Perform ? PutValue(_lref_, _rval_).
+        1. Return _rval_.
+      </emu-alg>
+      <emu-grammar>AssignmentExpression : LeftHandSideExpression `??=` AssignmentExpression</emu-grammar>
+      <emu-alg>
+        1. Let _lref_ be the result of evaluating |LeftHandSideExpression|.
+        1. Let _lval_ be ? GetValue(_lref_).
+        1. If _lval_ is neither *undefined* nor *null*, return _lval_.
+        1. If IsAnonymousFunctionDefinition(|AssignmentExpression|) and IsIdentifierRef of |LeftHandSideExpression| are both *true*, then
+          1. Let _rval_ be NamedEvaluation of |AssignmentExpression| with argument GetReferencedName(_lref_).
+        1. Else,
+          1. Let _rref_ be the result of evaluating |AssignmentExpression|.
+          1. Let _rval_ be ? GetValue(_rref_).
+        1. Perform ? PutValue(_lref_, _rval_).
+        1. Return _rval_.
+      </emu-alg>
       <emu-note>
-        <p>When an assignment occurs within strict mode code, it is a runtime error if _lref_ in step <emu-xref href="#step-assignmentexpression-evaluation-simple-putvalue"></emu-xref> of the first algorithm or step <emu-xref href="#step-assignmentexpression-evaluation-compound-putvalue"></emu-xref> of the second algorithm is an unresolvable reference. If it is, a *ReferenceError* exception is thrown. The |LeftHandSideExpression| also may not be a reference to a data property with the attribute value { [[Writable]]: *false* }, to an accessor property with the attribute value { [[Set]]: *undefined* }, nor to a non-existent property of an object for which the IsExtensible predicate returns the value *false*. In these cases a *TypeError* exception is thrown.</p>
+        <p>When this expression occurs within strict mode code and the respective GetValue of _lref_ step is reached, it is a runtime error if _lref_ is an unresolvable reference. If it is, a *ReferenceError* exception is thrown. Additionally if the PutValue step is reached, the _lref_ may not be a reference to a data property with the attribute value { [[Writable]]: *false* }, to an accessor property with the attribute value { [[Set]]: *undefined* }, nor to a non-existent property of an object for which the IsExtensible predicate returns the value *false*. In these cases a *TypeError* exception is thrown.</p>
```

Comment:
Presumably whatever made the two PutValue steps worth linking to before holds now, perhaps even more so, given the addition of new assignment forms. Mind adding `[id=]`s and enumerating the places where the ReferenceError is thrown in strict mode?

Reply:
Done.